lib/common/variadic/uint64OrHash.go

```diff
@@ -11,72 +11,72 @@ import (
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
-// Uint64OrHash represents an optional interface type (int,hash).
-type Uint64OrHash struct {
+// Uint32OrHash represents an optional interface type (int,hash).
+type Uint32OrHash struct {
 	value interface{}
 }
 
-// NewUint64OrHash returns a new variadic.Uint64OrHash given an int, uint64, or Hash
-func NewUint64OrHash(value interface{}) (*Uint64OrHash, error) {
+// NewUint32OrHash returns a new variadic.Uint32OrHash given an int, uint32, or Hash
```

Comment:
Should we rename this to `NewIntUint32Hash`? Sounds strange not to include `int` in its name :thinking:

Reply:
it's supposed to only be uint32 or hash, not int :p so I think it's fine to leave

Reply:
Should we remove the int type check in there then?